subworkflows/local/utils_nfcore_mcmicro_pipeline/main.nf

```diff
@@ -84,35 +84,44 @@ workflow PIPELINE_INITIALISATION {
     // Create channel from input file provided through params.input
     //
     def input_type
-
     if (input_sample) {
-        // FIXME toString works around nextflow#4944 - remove when fixed
-        sample_sheet_index_map = make_sample_sheet_index_map(input_sample.toString())
+        input_type = "sample"
         ch_samplesheet = Channel.fromSamplesheet(
             "input_sample",
-            //parameters_schema: parameters_schema,
             skip_duplicate_check: false
         )
-        .map { make_ashlar_input_sample(it, sample_sheet_index_map) }
+        .tap { ch_raw_samplesheet }
+        .map { validateInputSamplesheetRow(it, input_type) }
```

Comment:
I removed `validateInputSamplesheetRow` because it was currently only used for the `cycle_images` concept.
I like your idea about separate functions.  If/when we bring back row based validation I'll implement it as separate functions.